lib/ram_pwrdn/ram_pwrdn.c

```diff
@@ -3,118 +3,146 @@
  *
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
-#include <zephyr.h>
+
+#include <hal/nrf_power.h>
 #include <logging/log.h>
+#include <sys/util.h>
+#include <zephyr.h>
 
+#include <stdint.h>
 
 LOG_MODULE_REGISTER(ram_pwrdn, CONFIG_RAM_POWERDOWN_LOG_LEVEL);
 
-#if defined CONFIG_BOARD_NRF52840DK_NRF52840 || \
-	defined CONFIG_BOARD_NRF52833DK_NRF52833
+#if !defined(CONFIG_SOC_NRF52840) && !defined(CONFIG_SOC_NRF52833)
```

Comment:
Kconfig should take care of this, maybe this can be removed?

Reply:
True, but I just wanted to make it easier for a reader to know that it's for nRF52 only without looking for that info in Kconfig